### DISCUSSION TITLE
Sidebar only for Archive

### DISCUSSION THREAD

**Discussion by darkfrizzi** [bootscore/bootscore#766] [Q&A] [answered]
I would like to understand the best way to handle this.
If I put something in the sidebar widgets, it basically activates the sidebar throughout the site.

But if I would like to have it active only in the archives (shop, product category), what do I do?

I've seen that some have used a function, but then changed the cell display. Is there a way to upload it only where we want?

**Comment by crftwrk** [comment, marked as answer]
> If I put something in the sidebar widgets, it basically activates the sidebar throughout the site.

Exactly. A simple way is to add widgets to the sidebar and choosing a template without sidebar for posts and pages. But this won't work for WooCommerce. There you have to edit `woocommerce.php` as described in https://github.com/orgs/bootscore/discussions/630#discussioncomment-7962116.

But best way is to use a plugin to handle sidebar widgets contextual instead changing code.

It depends if you are on Bootscore v6 or v5. v6 uses block widgets and the only plugin which is able to handle block widgets is https://wordpress.org/plugins/widget-options/ so far. This plugin is active on https://bootscore.me, you can see it in action if you go to blog and shop where different widgets are displayed while the single product page has no sidebar (no widget active). 

Note that this plugin enables classic widgets on installation and you have to reenable block widgets in plugin settings. This plugin is heavy, but works fine.

If you still are on Bootscore v5, then you can use https://wordpress.org/plugins/widget-context/, which does nearly the same as Widget Options, but it is lighter and can only handle classic widgets.

Solved?

**Reply by darkfrizzi** [reply]
really perfect answer. Thank you, I understand.
Now I see how to solve it. I would like to keep the site as light as possible.
anyway yes I have v6, I only discovered your fantastic work yesterday

I don't know if I can ask it here in case I open another thread. But how did you make the color variations visible in the archives? did you use a plugin or something?

ty so much

**Reply by crftwrk** [reply]
> But how did you make the color variations visible in the archives?

Do you mean the dark mode? This is a plugin https://bootscore.me/shop/plugins/bs-dark-mode/  and visible everywhere, not only in archives.

**Reply by darkfrizzi** [reply]
no I meant, in the preview of the products, which are in the slider, the colors and sizes are already visible. To make the attributes visible did you use a plugin? Or is there a shortcode that I haven't seen. thanks

**Reply by crftwrk** [reply]
Ah, got it. You mean colors and sizes of the shirts. This is a WooCommerce hook, please open a new discussion, because other users may have the same question. I will share the code there.
